src/Randomizer.SMZ3.Tracking/VoiceCommands/ChatIntegrationModule.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Diagnostics;
+using System.Text.RegularExpressions;
+
+using Microsoft.Extensions.Logging;
+
+using Randomizer.SMZ3.ChatIntegration;
+
+namespace Randomizer.SMZ3.Tracking.VoiceCommands
+{
+    /// <summary>
+    /// Provides Tracker with stream chat integration.
+    /// </summary>
+    public class ChatIntegrationModule : TrackerModule, IDisposable
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see
+        /// cref="ChatIntegrationModule"/> class with the specified
+        /// dependencies.
+        /// </summary>
+        /// <param name="tracker">The tracker instance to use.</param>
+        /// <param name="chatClient">The chat client to use.</param>
+        /// <param name="logger">Used to write logging information.</param>
+        public ChatIntegrationModule(Tracker tracker, IChatClient chatClient, ILogger<ChatIntegrationModule> logger)
+            : base(tracker, logger)
+        {
+            ChatClient = chatClient;
+            ChatClient.Connected += ChatClient_Connected;
+            ChatClient.MessageReceived += ChatClient_MessageReceived;
+        }
+
+        /// <summary>
+        /// Gets the client used to integrate with chat.
+        /// </summary>
+        protected IChatClient ChatClient { get; }
+
+        /// <summary>
+        /// Frees up resources used by the <see cref="ChatIntegrationModule"/>.
+        /// </summary>
+        public void Dispose()
+        {
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Frees up resources used by the <see cref="ChatIntegrationModule"/>.
+        /// </summary>
+        /// <param name="disposing">
+        /// <see langword="true"/> when called from <see cref="Dispose()"/>;
+        /// otherwise, <see langword="false"/>.
+        /// </param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                ChatClient.Connected -= ChatClient_Connected;
+                ChatClient.MessageReceived -= ChatClient_MessageReceived;
+            }
+        }
+
+        private void ChatClient_MessageReceived(object sender, MessageReceivedEventArgs e)
+        {
+            var stopwatch = Stopwatch.StartNew();
+            try
+            {
+                foreach (var recognizedGreeting in Tracker.Responses.Chat.RecognizedGreetings)
```

Comment:
So something we may have to be worried about is chat spamming a ton and Tracker basically not shutting up because of it. I'm not sure exactly the best way to counter this, though. 

One thought is to maybe not process individual messages like this, but instead queue them up and process them in batches based on preferences. That way it could maybe say hello to multiple people in one go if there are multiple or even fall back to something like "hi everyone" if it's spammed ridiculously.

Another thought is maybe a quick way to toggle Twitch interaction on the fly by voice command and/or UI interactions on the tracker window. This would probably be good regardless of if the batch processing is done or not.

Reply:
Good point, although batching them would make this needlessly complicated. I could definitely limit it to once per user, though. Adds another opportunity for sass if they say it twice (and then ignoring it completely on subsequent attempts)